source/ShopPanel.cpp

```diff
@@ -345,14 +345,17 @@ void ShopPanel::DrawDetailsSidebar()
 		shipInfo.Update(*selectedShip, player.StockDepreciation(), player.GetDate().DaysSinceEpoch());
 
 		const Sprite *shipSprite = selectedShip->GetSprite();
+		float spriteScale = 1.;
+		if (shipSprite->Width() > (INFO_SIDE_WIDTH - 16))
+			spriteScale = (INFO_SIDE_WIDTH - 16) / shipSprite->Width();
 
 		Point itemPoint(point.X(), point.Y());
 		Point center(panelCenter, itemPoint.Y() + shipSprite->Height() / 2);
 		Point attrPoint(itemPoint.X(), itemPoint.Y() + shipSprite->Height());
 		Point outfPoint(attrPoint.X(), attrPoint.Y() + shipInfo.AttributesHeight());
 		Point descPoint(outfPoint.X(), outfPoint.Y() + shipInfo.OutfitsHeight());
 
-		SpriteShader::Draw(shipSprite, center);
+		SpriteShader::Draw(shipSprite, center, spriteScale);
```

Comment:
You should also swizzle the selected ship according to the player's swizzle.

Reply:
Thank you, solved with https://github.com/endless-sky/endless-sky/pull/5233/commits/093065f2f31b83b9402a4b623fcec05722660824.